File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/plugin/PluginDeliveryPage.java
package com.mieai.qqbot.persistence.plugin;

import java.util.List;
import java.util.Objects;
import java.util.Optional;

/** One immutable page of plugin deliveries and an optional next-page cursor. */
public record PluginDeliveryPage(
        List<PluginDelivery> deliveries,
        Optional<String> nextCursor) {

    public PluginDeliveryPage {
        deliveries = List.copyOf(Objects.requireNonNull(deliveries, "deliveries must not be null"));
        nextCursor = Objects.requireNonNull(nextCursor, "nextCursor must not be null");
    }
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/plugin/PluginDeliveryPage.kt
package com.mieai.qqbot.persistence.plugin

/** One immutable page of plugin deliveries and an optional next-page cursor. */
data class PluginDeliveryPage private constructor(
    val deliveries: List<PluginDelivery>,
    val nextCursor: String?,
    private val normalized: Boolean,
) {
    constructor(deliveries: List<PluginDelivery>, nextCursor: String?) : this(
        deliveries.toList(),
        nextCursor,
        true,
    )
}
